doc/md: v0.20 blog post (#2619)
ariga/atlas: doc/website/blog/2024-03-15-atlas-v-0-20.mdx

---
title: "Announcing v0.20: New Pricing Model, Django Support, Postgres Extensions, and more"
authors: hilakashai
tags: [release, postgresql, django, sql-server, ]
---

Hi everyone,

It's been a few weeks since our last version announcement and today I'm happy to share with you v0.20, which includes
some big changes and exciting features:

* [__New Pricing Model__](#new-pricing-model) - As we announced earlier this month, beginning today (March 15th)
the new pricing model will take effect. The new pricing is usage-based, offering you more flexibility and cost
efficiency. Read about what [prompted this change](/blog/2024/03/03/pricing-change) and view the new pricing plans [here](https://atlasgo.cloud/pricing).
* [__Django ORM Integration__](#django-orm-integration) -  Atlas now supports [Django](https://www.djangoproject.com)! Django is a  popular ORM for Python. Developers using either ORM can now use Atlas to automatically plan schema migrations for them, based on the desired state of their schema instead of crafting them by hand.

> * [__Support for PostgreSQL Extensions__](#support-for-postgresql-extensions) - Atlas now supports adding PostgreSQL extensions to your schema file to be loaded into the database.

Reviewer: ```suggestion
* [__Support for PostgreSQL Extensions__](#support-for-postgresql-extensions) - Atlas now supports installing and managing PostgreSQL extensions. 
```

> * [__Dashboards in the Cloud__](#dashboards-in-the-cloud) - The dashboard (previously the 'Projects' page) got a whole new look in Atlas Cloud. Now you can view the state of your projects at a glance.

Reviewer: ```suggestion
* [__Dashboards in the Cloud__](#dashboards-in-the-cloud) - The dashboard (previously the 'Projects' page) got a whole new look in Atlas Cloud. Now you can view the state of your projects and environments at a glance.
```

It can handle a wide range of schema changes, however, having been created in 2014, a very different era in software engineering,
it naturally has some limitations.

Some of the limitations of Django's migration system include:

1. **Database Features** - Because it was created to provide interoperability across database engines, Django's migration system
is centered around the "lowest common denominator" of database features.

2. **Ensuring Migration Safety** -  Migrations are a risky business. If you're not careful, you can easily cause data loss
or a production outage. Django's migration system does not provide a native way to ensure that a migration is safe to apply.

3. **Modern Deployments** - Django does not provide native integration with modern deployment practices such as GitOps or Infrastructure-as-Code.

Atlas, on the other hand, lets you manage your Django applications using the Database Schema-as-Code paradigm. This means that you can use
Atlas to automatically plan schema migrations for your Django project, and then apply them to your database.

Read the [full guide](/guides/orms/django) to set up Atlas for your Django project.

## Support for PostgreSQL Extensions

> [Postgres extensions](https://www.postgresql.org/download/products/6-postgresql-extensions/) are add-on modules that

Reviewer: ```suggestion
[Postgres extensions](https://www.postgresql.org/docs/current/sql-createextension.html) are add-on modules that
```

> enhance the functionality of the database. Extensions provide features that aren't included in the core Postgres system,
> such as additional data types, functions, and more.

Reviewer: ```suggestion
enhance the functionality of the database by introducing such as functions, data types, operators, and more.
```

> Support for adding these extensions to Atlas has been highly requested, so we are excited to announce that they are finally available!

Reviewer: ```suggestion
The support for extensions has been highly requested, so we are excited to announce that they are finally available!
```

To load an extension, add the extension block to your [schema file](/atlas-schema/hcl). For example, adding [PostGIS](https://postgis.net/) would look similar to:

```hcl title=schema.hcl
extension "postgis" {
 schema  = schema.public
 version = "3.4.1"
 comment = "PostGIS geometry and geography spatial types and functions"
}
```

> Read more about configuring extensions in your schema file [here](/atlas-schema/hcl#extension).

Reviewer: ```suggestion
Read more about configuring extensions in your schema [here](/atlas-schema/hcl#extension).
```

## Dashboards in the Cloud

[Atlas Cloud](https://atlasgo.cloud) has a new and improved dashboard view!

> When working with multiple databases, environments, or even schemas - it becomes increasingly difficult to

Reviewer: ```suggestion
When working with multiple databases, environments, or even projects - it becomes increasingly difficult to
```
